api/src/api.ts

```diff
@@ -190,6 +190,38 @@ export const plugin = {
       },
     });
 
+    /**
+     * The default implementation above is enormously wasteful in terms of space,
+     * due to a quadratic blowout in identical data.
+     * This route exists as a temporary fixture during development to dedupe data,
+     * accomplished via surgical DB queries rather than exhaustive ones.
+     */
+    server.route({
+      method: "GET",
+      path: "papers/{paperSelector}/entities-deduped",
+      handler: async (request) => {
+        const paperSelector = parsePaperSelector(request.params.paperSelector);
+        // Runtime type-checked during validation.
+        const entityTypes = request.query.type as EntityType[];
+        let res;
+        try {
+          res = await dbConnection.getDedupedEntitiesForPaper(paperSelector, entityTypes);
+        } catch (e) {
+          console.log(e);
```

Comment:
The above implementation doesn't do this, but it's something I've added in my own branch so you might also want to do it here. If you add an `h` argument to this handler function, you can `return h.response().code(500);` in here to make the API server respond with a 500 error instead of an empty set of data.

Reply:
👍